integration_tests/mockApis/crimeMatching/subjects.ts

```diff
@@ -1,44 +1,98 @@
 import { stubFor } from '../wiremock'
 
-// This has to match the path of the EM_CRIME_MATCHING_API_URL env variable
 const baseUrl = '/crime-matching'
 
-type StubSubjectSearchOptions = {
+type StubCreateSubjectsQuery200Options = {
+  status: 200
+  response: {
+    queryExecutionId: string
+  }
+}
+
+type StubSubjectsQuery400Options = {
+  status: 400
+  response: Array<{
+    field: string
+    message: string
+  }>
+}
+
+type StubCreateSubjectsQuery500Options = {
+  status: 500
+  response: string
+}
+
+type StubCreateSubjectQueryOptions =
+  | StubCreateSubjectsQuery200Options
+  | StubSubjectsQuery400Options
+  | StubCreateSubjectsQuery500Options
+
+// Default options returns a successful response with a mock queryExecutionId
+const defaultCreateSubjectOptions: StubCreateSubjectQueryOptions = {
+  status: 200,
+  response: {
+    queryExecutionId: '1234',
+  },
+}
+
+const stubCreateSubjectsQuery = (options: StubCreateSubjectQueryOptions = defaultCreateSubjectOptions) =>
+  stubFor({
+    request: {
+      method: 'POST',
+      urlPattern: `${baseUrl}/subjects-query`,
+    },
+    response: {
+      status: options.status,
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8',
+      },
+      jsonBody: options.response,
+    },
+  })
+
+type StubGetSubjects200Options = {
+  status: 200
   query: string
   response: Array<{
     nomisId: string
-    name: string | null
-    dateOfBirth: string | null
-    address: string | null
-    orderStartDate: string | null
-    orderEndDate: string | null
-    deviceId: string | null
-    tagPeriodStartDate: string | null
-    tagPeriodEndDate: string | null
+    name: string
+    dateOfBirth: string
+    address: string
+    orderStartDate: string
+    orderEndDate: string
+    deviceId: string
+    tagPeriodStartDate: string
+    tagPeriodEndDate: string
   }>
 }
 
-// Default options will return an empty array for any query string e.g. ?term=abc
-const defaultOptions: StubSubjectSearchOptions = {
+type StubGetSubjects404Options = {
```

Comment:
```suggestion
type StubGetSubjectsQuery404Options = {
```
Maybe? Just to keep consistency. Because we want to `get` the `SubjectsQuery` resource?